backend/market/models.py

```diff
@@ -8,51 +8,74 @@
 
 class Offer(models.Model):
     class Meta:
-        constraints = [models.UniqueConstraint(fields=["user", "sublet"], name="unique_offer")]
+        constraints = [models.UniqueConstraint(fields=["user", "item"], name="unique_offer")]
 
     user = models.ForeignKey(User, on_delete=models.CASCADE, related_name="offers_made")
-    sublet = models.ForeignKey("Sublet", on_delete=models.CASCADE, related_name="offers")
+    item = models.ForeignKey("Item", on_delete=models.CASCADE, related_name="offers")
     email = models.EmailField(max_length=255, null=True, blank=True)
     phone_number = PhoneNumberField(null=True, blank=True)
     message = models.CharField(max_length=255, blank=True)
     created_date = models.DateTimeField(auto_now_add=True)
 
     def __str__(self):
-        return f"Offer for {self.sublet} made by {self.user}"
+        return f"Offer for {self.item} made by {self.user}"
 
 
-class Amenity(models.Model):
+class Category(models.TextChoices):
+    SUBLET = "sublet"
+    APPLIANCE = "appliance"
+    COOKWARE = "cookware"
+    FOOD = "food"
+    BOOK = "book"
+    ELECTRONICS = "electronics"
+    FURNITURE = "furniture"
+    CLOTHING = "clothing"
+    TRANSPORTATION = "transportation"
+    ROOMDECOR = "roomdecor"
+    SPORTS = "sports"
+    TICKETS = "tickets"
+    GIFTCARD = "giftcard"
+    OTHER = "other"
+
+
+class Tag(models.Model):
     name = models.CharField(max_length=255, primary_key=True)
 
     def __str__(self):
         return self.name
 
 
-class Sublet(models.Model):
-    subletter = models.ForeignKey(User, on_delete=models.CASCADE)
-    sublettees = models.ManyToManyField(
-        User, through=Offer, related_name="sublets_offered", blank=True
+class Item(models.Model):
+    seller = models.ForeignKey(User, on_delete=models.CASCADE)
```

Comment:
I originally removed it since I thought that it was unnecessary since Offer is an explicitly defined intermediate table for that relation. However, I think I'll add it back since it makes it easier to directly make queries to the buyers of an item.

Reply:
Sounds good. Yea, using "through" models is always technically redundant so the reason its used is for sake of ease for syntax